pkg/util/mount/mount_linux.go

```diff
@@ -1171,3 +1215,111 @@ func doSafeOpen(pathname string, base string) (int, error) {
 
 	return finalFD, nil
 }
+
+// searchMountPoints finds all mount references to the source, returns a list of
+// mountpoints.
+// Some filesystems may share a source name, e.g. tmpfs. And for bind mounting,
+// it's possible to mount a non-root path of a filesystem, so we need to use
+// root path and major:minor to represent mount source uniquely.
+// This implementation is shared between Linux and NsEnterMounter
+func searchMountPoints(source, kubeletpath string) ([]string, error) {
+	mis, err := listMountInfos()
+	if err != nil {
+		return nil, err
+	}
+
+	rootPath := ""
+	majorMinor := ""
+
+	// Finding the underlying root path and major:minor if source is a mountpoint.
+	// We need search in backward order because it's possible for later mounts
+	// to overlap earlier mounts.
+	for i := len(mis) - 1; i >= 0; i-- {
+		if source == mis[i].mountPoint {
+			rootPath = mis[i].root
+			majorMinor = mis[i].majorMinor
+			break
+		}
+	}
+
+	// Get root path and major:minor from filesystem if we could not get from mounts.
+	if rootPath == "" || majorMinor == "" {
+		rootPath, majorMinor, err = getRootPathAndMajorMinor(kubeletpath)
```

Comment:
Good idea! Probably `getRootPathAndMajorMinor` is not necessary actually, because we can get mount references of device by comparing it with `source` field of mount, like it did previously.

Reply:
Updated in https://github.com/kubernetes/kubernetes/pull/62102/commits/ce9261eb64d2c3086b3d0bcc41d02cf2da7d0e3e. `getRootPathAndMajorMinor` and related functions are removed. Code is much simpler.